src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -1170,6 +1170,91 @@ int RadosStore::remove_topic_v2(const std::string& topic_name,
                                            params, objv_tracker, y, dpp);
 }
 
+int RadosStore::update_bucket_topic_mapping(const rgw_pubsub_topic& topic,
+                                            rgw::sal::Bucket* bucket,
+                                            bool add_mapping,
+                                            optional_yield y,
+                                            const DoutPrefixProvider* dpp) {
+  const auto& bucket_str =
+      rgw_make_bucket_entry_name(bucket->get_tenant(), bucket->get_name());
+  bufferlist empty_bl;
+  librados::ObjectWriteOperation op;
+  int ret = 0;
+  if (add_mapping) {
+    std::map<std::string, bufferlist> mapping{{bucket_str, empty_bl}};
+    op.omap_set(mapping);
+  } else {
+    std::set<std::string> to_rm{{bucket_str}};
+    op.omap_rm_keys(to_rm);
+  }
+  ret = rgw_rados_operate(dpp, rados->get_notif_pool_ctx(),
+                          get_bucket_topic_mapping_oid(topic), &op, y);
+  const auto str = add_mapping ? "add" : "remove";
```

Comment:
please add the ternary expression into the `ldpp_dout` line, so it is not called based on log level

Reply:
> please add the ternary expression into the `ldpp_dout` line, so it is not called based on log level

the ternary expression is used across 2 `ldpp_dout` places, and its needed for both log_levels.

Reply:
right, but these calls to `ldout_dpp` are mutually excluseive. and anyway, currently we always do these string operations, and if embedded insidte the log command they will be done only if log level is 20 or if the rados operation failed.

Reply:
done.